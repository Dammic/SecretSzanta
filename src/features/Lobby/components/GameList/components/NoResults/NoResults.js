import React from 'react'

import styles from './NoResults.css'

export const NoResults = () => (
    <div className={styles.noResults}>
        <span className={styles.noResultsHeadline}>
            No rooms found at this moment
        </span>
        <span className={styles.noResultsParagraph}>
            {"Click on the 'New Room' button to your right to create a new room"}
        </span>
    </div>
)
