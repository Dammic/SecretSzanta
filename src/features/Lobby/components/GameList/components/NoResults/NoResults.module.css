@import 'variables/variables.css';

.noResults {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    height: 100%;
    width: 100%;
    text-align: center;
}

.noResultsHeadline  {
    font-size: 24px;
    line-height: 28px;
    font-weight: bold;
    margin-bottom: 10px;
    color: #edd4d4;

    @media (--tablet) {
        font-size: 28px;
        line-height: 34px;
    }
}

.noResultsParagraph {
    font-size: 16px;
    font-weight: normal;
    line-height: 20px;
    color: #999;
    margin-bottom: 300px;
}


.noResultsHeadline, .noResultsParagraph {
    margin-left: 20px;
    margin-right: 20px;
}
