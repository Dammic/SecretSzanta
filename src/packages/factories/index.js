export { arrayOf } from './arrayOf'
export { aMessage } from './messages'
export { aRoom, aRoomsList, aPlayer, aPlayersList } from './lobby'
