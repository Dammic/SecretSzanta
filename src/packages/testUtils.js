import renderer from 'react-test-renderer';
import ShallowRenderer from 'react-test-renderer/shallow';

const shallowRenderer = new ShallowRenderer()

export const expectMatchingSnapshot = (element) => {
    const renderedElement = renderer.create(element)
    expect(renderedElement).toMatchSnapshot()
    renderedElement.unmount()
}

export const expectShallowMatchingSnapshot = (element) => {
    const renderedElement = shallowRenderer.render(element)
    expect(renderedElement).toMatchSnapshot()
    renderedElement.unmount()
}
