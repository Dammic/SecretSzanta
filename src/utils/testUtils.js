import { mount } from 'enzyme'

export const expectMatchingSnapshot = (element) => {
    const renderedComponent = mount(element)
    expect(renderedComponent).toMatchSnapshot()
}
