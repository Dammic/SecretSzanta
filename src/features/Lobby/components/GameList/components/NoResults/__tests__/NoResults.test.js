import React from 'react'
import { expectShallowMatchingSnapshot } from 'packages/testUtils'
import { NoResults } from '../NoResults'

describe('<NoResults />', () => {
    it('matches snapshot', () => {
        expectShallowMatchingSnapshot(<NoResults />)
    })
})
