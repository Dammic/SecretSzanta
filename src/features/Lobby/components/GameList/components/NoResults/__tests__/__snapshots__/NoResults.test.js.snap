// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NoResults /> matches snapshot 1`] = `
<div
  className="noResults"
>
  <span
    className="noResultsHeadline"
  >
    No rooms found at this moment
  </span>
  <span
    className="noResultsParagraph"
  >
    Click on the 'New Room' button to your right to create a new room
  </span>
</div>
`;
